hack/update-codegen.sh

```diff
@@ -14,104 +13,820 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Please do not add any logic to this shell script. Add logic to the go code
-# that generates the set-gen program.
+# shellcheck disable=2046 # printf word-splitting is intentional
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
+# This tool wants a different default than usual.
+KUBE_VERBOSE="${KUBE_VERBOSE:-1}"
+
 KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
 kube::golang::setup_env
 
-GO111MODULE=on GOPROXY=off go install k8s.io/kubernetes/pkg/generated/openapi/cmd/models-schema
-GO111MODULE=on GOPROXY=off go install k8s.io/code-generator/cmd/client-gen
-GO111MODULE=on GOPROXY=off go install k8s.io/code-generator/cmd/lister-gen
-GO111MODULE=on GOPROXY=off go install k8s.io/code-generator/cmd/informer-gen
-GO111MODULE=on GOPROXY=off go install k8s.io/code-generator/cmd/applyconfiguration-gen
-
-modelsschema=$(kube::util::find-binary "models-schema")
-clientgen=$(kube::util::find-binary "client-gen")
-listergen=$(kube::util::find-binary "lister-gen")
-informergen=$(kube::util::find-binary "informer-gen")
-applyconfigurationgen=$(kube::util::find-binary "applyconfiguration-gen")
-
-IFS=" " read -r -a GROUP_VERSIONS <<< "${KUBE_AVAILABLE_GROUP_VERSIONS}"
-GV_DIRS=()
-for gv in "${GROUP_VERSIONS[@]}"; do
-  # add items, but strip off any leading apis/ you find to match command expectations
-  api_dir=$(kube::util::group-version-to-pkg-path "${gv}")
-  nopkg_dir=${api_dir#pkg/}
-  nopkg_dir=${nopkg_dir#vendor/k8s.io/api/}
-  pkg_dir=${nopkg_dir#apis/}
-
-
-  # skip groups that aren't being served, clients for these don't matter
-  if [[ " ${KUBE_NONSERVER_GROUP_VERSIONS} " == *" ${gv} "* ]]; then
-    continue
-  fi
-
-  GV_DIRS+=("${pkg_dir}")
+DBG_CODEGEN="${DBG_CODEGEN:-0}"
```

Comment:
Is there a reason we need a new debug var as opposed to the standard KUBE_VERBOSE?

Reply:
I retained it from the Makefile.  As it is, setting KUBE_VERBOSE generates THOUUUUUUUUSANDS of lines of logging from the tools themselves, which isn't "helpful".